app/internal/wm/GioView.java

```diff
@@ -132,6 +134,72 @@ private void setCursor(int id) {
 		setPointerIcon(pointerIcon);
 	}
 
+    private void setStatusColor(int color, int luminance) {
```

Comment:
Maybe both `setStatusColor` and  `setNavigationColor` can be replace by `setColor(bar String, ....)`, in order to re-use all the code. It may simplify the code, but adds a lot of `if (bar == "navigation)`, since some constants are only available in some versions (`WindowInsetsController.APPEARANCE_LIGHT_NAVIGATION_BAR`, for instance).

Reply:
Java constants are resolved at compile time, so they don't need version guards.

Reply:
I change it to a single function `setBarColor` to reduce the duplicated code. 🎉